core/bin/snapshots_creator/src/main.rs

```diff
@@ -27,7 +30,7 @@ mod tests;
 
 async fn maybe_enable_prometheus_metrics(
     stop_receiver: watch::Receiver<bool>,
-) -> anyhow::Result<()> {
+) -> anyhow::Result<Option<JoinHandle<anyhow::Result<()>>>> {
```

Comment:
Nit: This function doesn't need to be `async`, and the outer `anyhow::Result<_>` is obsolete as well. (AFAIU, both of these could be caught with Clippy lints that are not enabled by default; you may try copying them from [another crate](https://github.com/matter-labs/zksync-era/blob/4fe284a273a0299528c00e01264a1c1c32f630c5/core/lib/merkle_tree/src/lib.rs#L41-L47) to see how many lints we're missing 🙃)

Reply:
I'd leave it as it for now. I want to play with clippy settings after company-wide holidays and fix found issues in this crate